ROVU/Core/src/utility/SimulatorMonitor.java

```diff
@@ -1,31 +1,83 @@
 package utility;
 
+import java.awt.geom.Rectangle2D;
 import java.util.ArrayList;
+
 import java.util.List;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 
+import model.Environment;
+import model.Hospital;
 import model.Mission;
 import model.MissionPoint;
 import project.AbstractRobotSimulator;
 import project.AbstractSimulatorMonitor;
 import project.Point;
+import robot.A_Star;
+import model.Node;
 import robot.RobotHandler;
+import simbad.sim.Boundary;
 import simbad.sim.EnvironmentDescription;
+import simbad.sim.Wall;
 
 public class SimulatorMonitor extends AbstractSimulatorMonitor<RobotHandler> {
+	Point [] com ;
+	public  List <Boundary> bounds;
+	public  List <Wall> walls;
+
+	
 	Set<RobotHandler> r;
-	public SimulatorMonitor(Set<RobotHandler> robots, EnvironmentDescription e) {
+	public SimulatorMonitor(Set<RobotHandler> robots, EnvironmentDescription e,List <Boundary> bounds, List <Wall> walls) {
 		super(robots, e);
 		r = robots;
+		this.walls=walls;
+		this.bounds=bounds;
 	}
 
-	
+
 
 	public void update(RobotHandler arg0) {
-//		System.out.println(arg0.getName());
-//		System.out.println(arg0.getPosition());
-//
+		
+		/*
+		RobotHandler [] arr = new RobotHandler [r.size()];
+		boolean check=true;
+		if (check){
+			check=false;
+		int i=0;
+		
+		for (RobotHandler h:r){
+			arr[i]=h;
+			i++;
+		}
+		
+		A_Star test = new A_Star();
+		Environment env = new Environment(new ArrayList<Node>());
+		env.setWalls(bounds, walls);
+		env.setCoefficient(1);
+
+		List <Node> list = env.generateEmptyGrid(20, 1);//20 is standard size -10 <> +10, 20/coefficient = number of nodes in line
+
+
+		test.setStartDestination(env.getMap().get(2), env.getMap().get(8));
```

Comment:
Are `.get(2)` and `.get(8)` just arbitrary numbers or...?

It's more clear if it was written like this:
``` Java
Node start = env.getMap().get(2);
Node finish = env.getMap().get(8);
test.init(start, finish); // I commented in the Node file to change the method name
```